Examples/HttpsExampleServer/Program.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Security;
+using System.Net.Sockets;
+using System.Reflection;
+using System.Security.Authentication;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Console;
+using Http2;
+using Http2.Hpack;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        var logProvider = new ConsoleLoggerProvider((s, level) => true, true);
+        // Create a TCP socket acceptor
+        var listener = new TcpListener(IPAddress.Any, 8889);
+        listener.Start();
+        Task.Run(() => AcceptTask(listener, logProvider)).Wait();
+    }
+
+    static bool AcceptIncomingStream(IStream stream)
+    {
+        Task.Run(() => HandleIncomingStream(stream));
+        return true;
+    }
+
+    static byte[] responseBody = Encoding.ASCII.GetBytes(
+        "<html><head>Hello World</head><body>Content</body></html>");
+
+    private static SslServerAuthenticationOptions options = new SslServerAuthenticationOptions()
+    {
+        // get our self signed certificate
+        ServerCertificate = new X509Certificate2(ReadWholeStream(Assembly.GetExecutingAssembly()
+            .GetManifestResourceStream("HttpsExampleServer.localhost.p12"))),
+        // this line adds ALPN, critical for HTTP2 over SSL
+        ApplicationProtocols = new List<SslApplicationProtocol>(){SslApplicationProtocol.Http2},
+        ClientCertificateRequired = false,
+        CertificateRevocationCheckMode = X509RevocationMode.NoCheck,
+        EnabledSslProtocols = SslProtocols.Tls12
+    };
+
+    static byte[] ReadWholeStream(Stream stream)
+    {
+        byte[] buffer = new byte[16 * 1024];
```

Comment:
Using `StreamReader.ReadToEnd` might have been possible too, but seems like that returns a `string` and therefore assumes an encoding.

Reply:
Bear in mind that this is only used for loading the self-signed certificate from the assembly, the rest of the code is exactly as it was in the ExampleServer project. I think to be encoding-safe we should keep this as is.

Reply:
Sure, that's fine. Btw: I would also be fine with just having one single example, where HTTPS is activated via a commandline argument (simliar to `CliClient`). Might be less duplicated code than 2 things which are nearly the same.

Reply:
Thanks for the try. I think I would prefer it without the latest change, as I think sharing only a  single method might be more confusing to readers.

Can you wo back to the last one and keep other improvements for later on? I'm not sure whether github lets me merge a previous commit.

Reply:
Done